### PR TITLE
adding merge function

### PR DESCRIPTION
**Pull request by rtorr**
@mattwondra also inspired this.

This merge function will give an easy way to combine objects, like extend, but will return a new object, rather than change the `destinationObject`.

### CODE REVIEW

File: modules/merge/merge.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var extend = require('amp-extend');
+
+module.exports = function merge() {
+    var args = [].slice.call(arguments).sort();
```
Comment (HenrikJoreteg) [inline modules/merge/merge.js]: I'm not clear on why you'd call `sort` here.

**Comment by HenrikJoreteg** [comment]
any thoughts on this method from @AmpersandJS/core-team ?

File: modules/merge/merge.js (hunk 1 above)
Comment (rtorr) [inline modules/merge/merge.js]: I think I was doing something in a test and it got left behind :/

Thanks.

**Comment by lukekarrys** [comment]
I like it. It's simple but there are often times in code when I want to express that I'm trying to do something different than extending an existing object. I'm a fan of this over `extend({}, obj, obj1)` since it's being more explicit.

+1

**Comment by rtorr** [comment]
@lukekarrys that is exactly our use case at work. We have parts of our code that we always want "merge". It betters the chance of being more clear and not leaving room for the mistake of leaving out the first object. We already implemented this in our project, and thought it might be useful.

**Comment by HenrikJoreteg** [comment]
agreed, i like to too, let's do it.

**Pull request merged**